### PR TITLE
feat: migrate `MutDeque` to use `struct` instead of `enum`

### PR DESCRIPTION
**Pull request by dghosef**
This relies on #8334 to be merged first, so it is a draft for now

#8308

### CODE REVIEW

File: main/src/library/MutDeque.flix [hunk 1]
```diff
@@ -441,29 +412,28 @@ mod MutDeque {
...
+        loop(d€front)
 
     ///
     /// Apply the effectful function `f` to all the elements in the MutDeque `d`
     /// along with that element's index.
     ///
     pub def forEachWithIndex(f: (Int32, a) -> Unit \ ef, d: MutDeque[a, r]): Unit \ { ef, r } = region rc {
-        let ix = Ref.fresh(rc, 0);
-        forEach(x -> { let i = Ref.get(ix); f(i, x); Ref.put(i + 1, ix) }, d)
+        let ix = ref 0 @ rc;
```
Comment (magnus-madsen) [inline main/src/library/MutDeque.flix:435]: Accidental re-introduction of ref.

Comment (magnus-madsen) [reply main/src/library/MutDeque.flix:435]: You also have a few deref's that you will want to fix.

File: main/src/library/MutDeque.flix [hunk 2]
```diff
@@ -106,8 +109,7 @@ mod MutDeque {
     /// Returns `true` if `d` is empty.
     ///
     pub def isEmpty(d: MutDeque[a, r]): Bool \ r =
-        let MutDeque(_, _, f, b) = d;
-        Ref.get(f) == Ref.get(b)
+        d€front == d€back
```
Comment (magnus-madsen) [inline main/src/library/MutDeque.flix:112]: Seeing the euro-symbol is so jarring 😆

File: main/src/library/MutDeque.flix [hunk 3]
```diff
@@ -132,37 +134,29 @@ mod MutDeque {
...
-        let back = Ref.get(back_);
         let c = capacity(d) - 1;
         def loop(i, e, acc) =
             if (i == e)
                 acc
             else
-                loop(Int32.bitwiseAnd(i + 1, c), e, f(acc, Array.get(i, ar)));
-        loop(front, back, s)
+                loop(Int32.bitwiseAnd(i + 1, c), e, f(acc, Array.get(i, d€values)));
+        loop(d€front, d€back, s)
```
Comment (magnus-madsen) [inline main/src/library/MutDeque.flix:143]: The simplification in code is pretty great though.

File: main/src/library/MutDeque.flix (hunk 2 above)
Comment (dghosef) [reply main/src/library/MutDeque.flix:112]: Haha, it's peak language design

**Comment by magnus-madsen** [comment]
@jaschdoc Could you do a super-quick review?

**Review by jaschdoc — APPROVED** [review]

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**